front/app/containers/Authentication/useSteps/stepConfig/signUpFlow.ts

```diff
@@ -23,17 +27,14 @@ import {
   UpdateState,
 } from '../../typings';
 import { Step } from './typings';
-import { UseMutateFunction } from '@tanstack/react-query';
-import { IUser, IUserUpdate } from 'api/users/types';
-import { CLErrorsWrapper } from 'typings';
+import { OnboardingType } from 'api/authentication/authentication_requirements/types';
 
 export const signUpFlow = (
   getAuthenticationData: () => AuthenticationData,
   getRequirements: GetRequirements,
   setCurrentStep: (step: Step) => void,
   updateState: UpdateState,
-  anySSOProviderEnabled: boolean,
-  updateUser: UseMutateFunction<IUser, CLErrorsWrapper, IUserUpdate>
```

Comment:
The bug was that we were using this `updateUser` function, which we were prop drilling down a few components. Because it's not an async function, we were requesting the new authentication requirements before the backend had processed the user update.

By importing the `updateUser` async function directly and using that instead, we can await it and only request the authentication requirements afterwards.